torch/utils/_config_module.py

```diff
@@ -177,6 +190,12 @@ def save_config_portable(self) -> Dict[str, Any]:
             config[key] = self._config[key]
         return config
 
+    def freeze_key(self, name: str) -> None:
```

Comment:
@eellison do you see any problem with this? TL;DR this PR has new functionality to "freeze" config settings that we don't support changing after they're first read. This is how that's accomplished.